### DISCUSSION TITLE
Release planning for PM 3.2.1

### DISCUSSION THREAD

**Discussion by Olf0** [sailfishos-patches/patchmanager#244] [General]
I am basically finished with what I intended to achieve for PM 3.2.1, only the updated translations from Transifex triggered by the big "strings rework"-PR #227 still have to come in (well, SV already has, because eson is always very quick, and I am taking care about the DE one and may adapt some easy things in the NL one) and the fix-up PR #240 for that still has to be merged.

Side note: In addition to the "strings rework", my other goal was to consolidate and generalise [the function `mangle_libpath` in `pm_apply`](https://github.com/sailfishos-patches/patchmanager/blob/master/src/tools/pm_apply#L82-L146).  While I have the impression that this worked well, including the subsequent optimisation by @nephros and our mutual reviews (which resulted in some back and forth, but ultimately in much better code), one thing I am slightly concerned about still is corresponding C++ code for 32-/64-bit mangling, [as detailed before](https://github.com/sailfishos-patches/patchmanager/issues/220#issuecomment-1001233951): I took a look at it and found nothing, but due to my slim C++ know-how and lack of experience that does not mean much; maybe you (@nephros and @b100dian) can spare the time to re-review you code once more, also under the aspect that `pm_apply` does cover many more cases than before (i.e.: multiple, different library paths on the mangle list to convert in a single Patch, multiple locations to convert the same mangle-able library path in a single Patch etc.).

If you have other issues or PRs which you want to be in PM 3.2.1, please assign them to the milestone "3.2.1".

Note that I have set the due date to 15. February 2022, which is just a goal (hence to be handled flexibly), but new issues or PRs assigned to the milestone "3.2.1" shall be realistically resolvable before 1. February, so they can be well reviewed and tested before the due date a fortnight later.  OTOH I do not mind to release earlier than 15. Feb., if appropriate.
Side note: I also intend to care about the process at SailfishOS:Chum for this release (so I  can become accustomed to it), if nobody else is keen on doing this.

**Comment by nephros** [comment]
No real PRs pending from my side. Milestone and date are fine by me.

SFOS 4.4 is coming RSN^TM, lets see if that breaks anything for us.

**Reply by Olf0** [reply]
Yes, we ~~should definitely~~ might wait to see, if SailfishOS 4.4.0 creates any fallout.

**Reply by Olf0** [reply]
It is obvious by now that there will not be a SailfishOS 4.4.0 release before February 15th, because there is no EA release yet (which is still out for at least a week, although not for many weeks like years ago).  And Jolla's "soon™", which was not even stated for this, may take any time, as we know.

So either we (basically I) release this or next weekend, which would free the master branch for merges again, or we keep on waiting.  I am inclined to release on this Sunday or next weekend, depending on the weather (for cycling), but if you have reasons or preferences for any of these or other dates, please let me know.
Opinions, @nephros and @b100dian?

**Comment by b100dian** [comment]
References: ***@***.***>
 ***@***.***>
 
Content-Type: text/plain; charset=utf-8
Content-Transfer-Encoding: base64
Date: Fri, 11 Feb 2022 08:20:12 +0000
Message-ID: ***@***.***>
MIME-Version: 1.0

QXMgc29vbiBhcyB5b3UgYXJlIGNvbWZvcnRhYmxlLA0KVGhhbmtzDQoNCk9uIEZyaWRheSwgRmVi
cnVhcnkgMTEsIDIwMjIsIG9sZiB3cm90ZToNCj4gSXQgaXMgb2J2aW91cyBieSBub3cgdGhhdCB0
aGVyZSB3aWxsIG5vdCBiZSBhIFNhaWxmaXNoT1MgNC40LjAgcmVsZWFzZSBiZWZvcmUgRmVicnVh
cnkgMTV0aCwgYmVjYXVzZSB0aGVyZSBpcyBubyBFQSByZWxlYXNlIHlldCAod2hpY2ggaXMgc3Rp
bGwgb3V0IGZvciBhdCBsZWFzdCBhIHdlZWssIGFsdGhvdWdoIG5vdCBmb3IgbWFueSB3ZWVrcyBs
aWtlIHllYXJzIGFnbykuICBBbmQgSm9sbGEncyAic29vbuKEoiIsIHdoaWNoIHdhcyBub3QgZXZl
biBzdGF0ZWQgZm9yIHRoaXMsIG1heSB0YWtlIGFueSB0aW1lLCBhcyB3ZSBrbm93Lg0KPiANCj4g
U28gZWl0aGVyIHdlIChiYXNpY2FsbHkgSSkgcmVsZWFzZSB0aGlzIG9yIG5leHQgd2Vla2VuZCwg
d2hpY2ggd291bGQgZnJlZSB0aGUgbWFzdGVyIGJyYW5jaCBmb3IgbWVyZ2VzIGFnYWluLCBvciB3
ZSBrZWVwIG9uIHdhaXRpbmcuICBJIGFtIGluY2xpbmVkIHRvIHJlbGVhc2Ugb24gdGhpcyBTdW5k
YXkgb3IgbmV4dCB3ZWVrZW5kLCBkZXBlbmRpbmcgb24gdGhlIHdlYXRoZXIgKGZvciBjeWNsaW5n
KSwgYnV0IGlmIHlvdSBoYXZlIHJlYXNvbnMgb3IgcHJlZmVyZW5jZXMgZm9yIGFueSBvZiB0aGVz
ZSBvciBvdGhlciBkYXRlcywgcGxlYXNlIGxldCBtZSBrbm93LiAgT3BpbmlvbnMsIEBuZXBocm9z
IGFuZCBAYjEwMGRpYW4/DQo+IA0KPiAtLSANCj4gUmVwbHkgdG8gdGhpcyBlbWFpbCBkaXJlY3Rs
eSBvciB2aWV3IGl0IG9uIEdpdEh1YjoNCj4gaHR0cHM6Ly9naXRodWIuY29tL3NhaWxmaXNob3Mt
cGF0Y2hlcy9wYXRjaG1hbmFnZXIvZGlzY3Vzc2lvbnMvMjQ0I2Rpc2N1c3Npb25jb21tZW50LTIx
NTQ2NTgNCj4gWW91IGFyZSByZWNlaXZpbmcgdGhpcyBiZWNhdXNlIHlvdSB3ZXJlIG1lbnRpb25l
ZC4NCj4gDQo+IE1lc3NhZ2UgSUQ6IDxzYWlsZmlzaG9zLXBhdGNoZXMvcGF0Y2htYW5hZ2VyL3Jl
cG8tZGlzY3Vzc2lvbnMvMjQ0L2NvbW1lbnRzLzIxNTQ2NThAZ2l0aHViLmNvbQ0KDQotLSANClNl
bnQgZnJvbSBteSBTYWlsZmlzaCBkZXZpY2U=

**Reply by Olf0** [reply]
Running that block of characters through `base64 -d` outputs:
> As soon as you are comfortable,
Thanks
…

P.S.: @b100dian, please check your mail client setup, this is looking awkward.

**Reply by b100dian** [reply]
Yeah, I should stop using that, it's the jolla-email with a google account, - thanks for deciphering it!

**Reply by Olf0** [reply]
K9 Mail (from F-Droid or Google Play Store) works fine under AlienDalvik and may do better.
Just a guess, I prefer GitHub's web-fronted, but Jolla's mail client is unbearable IMO.

**Comment by Olf0** [comment]
Well, so I "just did it":
1. Tagged 3.2.1: https://github.com/sailfishos-patches/patchmanager/releases/tag/3.2.1
2. Altered the value of the parameter "version" to the current tag `3.2.1` (from `3.2.0`) in [the `_service` file](https://build.sailfishos.org/package/view_file/sailfishos:chum:testing/patchmanager/_service?expand=1).
3. As I had the impression nothing was happening, I also hit ["`Trigger services`"](https://build.sailfishos.org/package/show/sailfishos:chum:testing/patchmanager). In hindsight this might have been superfluous.
    Is it really sufficient to solely alter the parameter "version" in the `_service` file in order to trigger downloading the corresponding `tar.gz` file, plus extracting and compiling its contents according to the spec file?
4. All packages were built successfully, see https://build.sailfishos.org/package/show/sailfishos:chum:testing/patchmanager
    I intend to test the one for SFOS4.0.1@ARMv7-A tomorrow.

**Reply by nephros** [reply]
> As I had the impression nothing was happening, I also hit ["Trigger services"](https://build.sailfishos.org/package/show/sailfishos:chum:testing/patchmanager). In hindsight this might have been superfluous.
Is it really sufficient to solely alter the parameter "version" in the _service file in order to trigger downloading the corresponding tar.gz file, plus extracting and compiling its contents according to the spec file?

Nice, welcome to OBS ;)

Yes, editing that field is enough to do a rebuild (the web interface however often does not update itself, often a reload of the page is needed to see changes).  
The tag must exist of course, AND the spec file should have the same (or matching) version. If they differ it will pull the code from the tag, but build a package named with the version from the spec file.

If you edit it to be empty, it will use the latest commit in the specified branch (or master if empty).

BTW, there is a relatively nice CLI tool `osc` available to interact with OBS, so you can edit the file and trigger a rebuild from the phone or elsewhere. With osc, such a rebuild would be:

     osc co sailfishos:chum:testing patchmanager  # get the obs project and package
     cd sailfishos:chum:testing/patchmanager
     $EDITOR _service                           # edit, edit, edit
     osc ci -n _service                         # upload edited file (triggers a rebuild)
     # the following are optional:
     osc service remoterun                      # manual trigger of a re-clone and rebuild
     osc blt <reponame> <arch name>             # watch the build log as it builds (buld-log-tail)
     osc getbinaries <reponame> <arch name>     # download build results

**Reply by Olf0** [reply]
Thank you very much for the reply, including a thorough answer, and how to achieve this via `osc`.

Actually installing `osc`and executing 4 to 7 commands does sound "nineties" to me, altering a single version string via a web-editor has much more appeal to me (even though I would have preferred the "nineties"-method ten years ago), now that I know that this is fully sufficient.

**Reply by nephros** [reply]
> Thank you very much for the reply, including a thorough answer, and how to achieve this via `osc`.
> 
> Actually installing `osc`and executing 4 to 7 commands does sound "nineties" to me, altering a single version string via a web-editor has much more appeal to me (even though I would have preferred the "nineties"-method ten years ago), now that I know that this is fully sufficient.

Preferences :) 

It's nice having that kind of interface when you're doing something on the terminal anyway (like programming). Also, I find it much easier to do when on the phone than using a browser.

**Reply by Olf0** [reply]
> Preferences :)

Choice!
It is paramount to have the choice between profoundly different routes to achieve something on / with a computer.

> It's nice having that kind of interface when you're doing something on the terminal anyway (like programming).

And I always assumed that is what IDEs are for!?!
But "I am not a dev", just to emit that idiot-phrase. :stuck_out_tongue_winking_eye:

> Also, I find it much easier to do when on the phone than using a browser.

Funny, that is also my verdict after more than a decade with Maemo and SailfishOS.
Most other smartphone users react with disbelief.
But since I started to dwell at home-office around the clock two years ago, the laptop is always up an running. :frowning_face:

**Comment by Olf0** [comment]
Done, see Patchmanager's releases [at GitHub](https://github.com/sailfishos-patches/patchmanager/releases/tag/3.2.1), at [OpenRepos](https://openrepos.net/content/patchmanager/patchmanager) and submitted a request at [SailfishOS:Chum OBS](https://build.sailfishos.org/request/show/1933).
Hope all is fine.

P.S.: This created quite some "fallout", i.e., issues and PRs at various projects.
